src/core/Akka/Actor/ActorPath.cs

```diff
@@ -202,70 +240,109 @@ internal IReadOnlyList<string> ElementsWithUid
         {
             get
             {
-                if (this is RootActorPath) return EmptyElements;
-                var elements = (List<string>)Elements;
-                elements[elements.Count - 1] = AppendUidFragment(Name);
-                return elements;
+                if (_depth == 0)
+                    return ImmutableArray<string>.Empty;
+
+                var b = ImmutableArray.CreateBuilder<string>(_depth);
+                b.Count = _depth;
+                var p = this;
+                for (var i = 0; i < _depth; i++)
+                {
+                    b[_depth - i - 1] = i > 0 ? p._name : AppendUidFragment(p._name);
+                    p = p._parent;
+                }
+                return b.ToImmutable();
```

Comment:
See above comments about .ToImmutable() vs List or MoveToImmutable().

Reply:
Yes, this is what I intended to do.